src/components/QrButton.tsx

```
@@ -0,0 +1,160 @@
+                      onClick={fetchQrCode}
+                      className="px-4 py-2 text-sm font-medium text-white bg-blue-600 rounded-md hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
+                    >
+                      Retry
+                    </button>
+                  </div>
+                ) : qrData ? (
+                  <div className="text-center">
+                    {/* QR Code SVG */}
+                    <div
+                      className="inline-block p-4 bg-white border border-gray-200 rounded-lg mb-4"
+                      dangerouslySetInnerHTML={{ __html: qrData.svg }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)